src/js/utils/element-utils.js

```diff
@@ -14,3 +14,9 @@ export const getAllSlotted = (el, name) => {
 };
 
 export const getSlotted = (el, name) => getAllSlotted(el, name)[0];
+
+export const containsWithShadow = (refNode, otherNode) => {
+  if (!refNode || !otherNode) return false;
+  if (refNode.contains(otherNode)) return true;
+  return containsWithShadow(refNode.shadowRoot, otherNode);
```

Comment:
Seems like the reason why it doesn't is that the shadow root is a Document Fragment, so, technically, the shadow DOM isn't in the main document DOM.
The other approach would've been to iterate through `parentNode` (and not `parentElement` because the DocumentFragment is a node), but this way is better and more targeted to our needs.